api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -301,6 +301,12 @@
      */
     int getId();
 
+    /**
+     * Return the number of windows before the next checkpoint including the current window.
+     * @return Number of windows from checkpoint, 1 if the checkpoint will be after the current window
+     */
+    int getWindowsFromCheckpoint();
```

Comment:
I think using an attribute is odd since this attribute would change all the time (every single window), as opposed to other attributes, which are mostly constant throughout the application execution.